pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -99,7 +99,10 @@ func (e *edgeFetcher) GetWatermark(inputOffset isb.Offset) processor.Watermark {
 	for _, p := range allProcessors {
 		debugString.WriteString(fmt.Sprintf("[Processor: %v] \n", p))
 		var t = p.offsetTimeline.GetEventTime(inputOffset)
-		if t != -1 && t < epoch {
+		if t == -1 { // watermark cannot be computed, perhaps a new processing unit was added or offset fell off the timeline
+			epoch = t
+			break
```

Comment:
> return -1 for watermark if we don't find an entry in the offset timeline store in any processor

The PR description doesn't match the logic here.

For example, if we have processors A B C, if processor A's watermark for inputOffset is -1, then we break the for loop and skip processors B and C. 
The description above means if all A B C returns -1 then we return -1.

Reply:
We already have the logic here 
https://github.com/numaproj/numaflow/pull/402/files#diff-6af877f3295078204591f48a2180bc855394b50997f764275b25ef2ffff678daL110-R116
to return -1 if no entry found in any offset timeline.

Reply:
updated the description

Reply:
But we need to set the overall watermark to -1 if watermark cannot be fetched for any processing unit

Reply:
> https://github.com/numaproj/numaflow/pull/402/files#diff-6af877f3295078204591f48a2180bc855394b50997f764275b25ef2ffff678daL110-R116

👌 

do we still need this logic then?https://github.com/numaproj/numaflow/blob/68631426b61b85fe72ca0571648dde4917d33a2e/pkg/watermark/fetch/edge_fetcher.go#L114-L116

Reply:
we don't need it

Reply:
I think we need it, if there are no processors, we will return int max